docs/paper/dev/api/components.md

```diff
@@ -0,0 +1,180 @@
+(click and hover). The other component types and these style options have poor or missing
+representations in the legacy string format.
+
+## Usage
+
+Representing text as Components is now the supported way of representing text for Paper and Velocity. They are used
+for almost all aspects of text being displayed to clients. Text like item names, lore, boss bars, team prefixes and
+suffixes, custom names, and much more all support Components in respective APIs.
+[Mojang has stated](https://bugs.mojang.com/browse/MC-190605?focusedId=993040&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-993040)
+that client support for the legacy format with `§` will be removed in the future.
+
+
+:::tip
+
+In the Paper API, there are lots of deprecated methods and types that deal with this legacy format. This is to
+signal that a better alternative in Components is available and should be migrated to going forward.
+
+:::
+
+## Components
+
+Components can be interacted with as objects. There are different interfaces for each type along with
+builders for all the types. These objects are immutable so when constructing more complex components, it's
+recommended to use builders to avoid creating new Component instances with every change.
+
+```java
+// This is a sub-optimimal construction of the
+// component as each change creates a new component
+final Component component = Component.text("Hello")
+    .color(TextColor.color(0x13f832))
+    .append(Component.text(" world!", NamedTextColor.GREEN));
+
+/* This is an optimal use of the builder to create
+   the same component. Also note that Adventure
+   Components are designed for use with static method imports 
+   to make code less verbose */
+final Component component = text()
+    .content("Hello").color(color(0x13f832))
+    .append(text(" world!", GREEN))
+    .build();
+```
+
+:::info
+
+For complete documentation on the Adventure Component API Paper and Velocity use, please look at the
+[Adventure documentation](https://docs.advntr.dev).
+
+:::
+
+## MiniMessage
+
+Paper and Velocity include the MiniMessage library which is a string representation of components. If you prefer working with
+strings rather than objects, MiniMessage is vastly superior to the legacy string format. It can utilize the tree
+structure for style inheritance and can represent the more complex component types while legacy cannot.
+
+```java
+final Component component = MiniMessage.miniMessage().deserialize(
+    "<#438df2><b>This is the parent component; its style is " +
+    "applied to all children.\n<u><!b>This is the first child, " +
+    "which is rendered after the parent</!b></u><key:key.inventory></b></#438df2>"
+);
+```
+
+We recommend using this format for user-facing input such as commands or configuration values.
+
+:::info
+
+MiniMessage is a part of Adventure, and you can find its documentation on [Adventure's documentation](https://docs.advntr.dev/minimessage/index.html).
+
+:::
+
```

Comment:
add a :::tip for the MiniMessage WebViewer located here: https://webui.advntr.dev/